lib/facility.rb

```diff
@@ -1,14 +1,59 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+  
+  def register_vehicle(vehicle)
+    if services.include?("Vehicle Registration")
+      vehicle.registration_date = Date.today
+      @registered_vehicles << vehicle
+
+      if vehicle.antique?
+        vehicle.plate_type = :antique
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle? == true
+        vehicle.plate_type = :ev
+        @collected_fees += 200
+      else vehicle.plate_type = :regular
+        @collected_fees += 100
+      end
+    end
+  end
+
+  def administer_written_test(registrant)
+    if services.include?("Written Test") && registrant.permit? == true && registrant.age >= 16
```

Comment:
A couple of thoughts:
1. For lines that get really long like this one, you could consider making a helper method.
2. You can omit the `== true` because `registrant.permit?` will already return a boolean.
```
def written_eligible?(registrant)
  services.include?("Written Test") && registrant.permit? && registrant.age >= 16
end
```